File: Day 3/Day 3/challenge_1.py
#Challenge 1 (Day 3)

#function to calculate sum of odd divisors
def odivsum(n):
    s = 0
    for i in range(1, n + 1):
        if n % i == 0 and i % 2 == 1:
            s += i
    return s

for t in range(int(input())):
    a, b = map(int,input().split())
    s = 0
    for i in range(a, b + 1):
        s += odivsum(i)
    print(s)
        
        

File: Day 3/Day 3/challenge_1.c
#include <stdio.h>


int main() {
int t;
    scanf("%d",&t);
    for( int i=0;i<t;i++)                             //Loop for each test case
    { int l,r,tsum=0;
        scanf("%d %d",&l,&r);                          //input l and r
        while(l<=r)                                   //while loop to get each element between l and r
        { int sum=0; 
           for(int j=1;j<=l;j++)                      
         if(((l%j)==0)&&((j%2)==1))                   // checking for divisors of l and which are odd 
            sum+=j;                                   // sum of each odd divisors
           
            tsum+=sum;                                 // The sum of each divisors of each number between l&r is added together
         l++;   
        }
      printf("%d \n",tsum);                             //output the total sum
    }
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */    
    return 0;
}
